modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/RuntimeMessage.java

```diff
@@ -32,9 +32,19 @@
 
     protected final String message;
 
-    public RuntimeMessage(Level level, String message) {
+    protected final Source source;
+
+    protected final String sourceId;
+
+    public RuntimeMessage(Level level, String message, Source source, String sourceId) {
         this.level = level;
         this.message = message;
+        this.source = source;
+        this.sourceId = sourceId;
+    }
+
+    public RuntimeMessage(Level level, String message) {
+        this(level, message, null, null);
```

Comment:
Could we deprecate this constructor?

Furthermore, can we pass `CODE` as source and this class name as sourceId to avoid future check on these fields due to deprecation?

Reply:
I removed the constructor since it was introduced recently.
I added an unknown source and fallback values for getters to avoid dealing with null values.